routers/web/repo/download.go

```diff
@@ -71,7 +71,8 @@ func ServeBlobOrLFS(ctx *context.Context, blob *git.Blob, lastModified time.Time
 				log.Error("ServeBlobOrLFS: Close: %v", err)
 			}
 		}()
-		return common.ServeData(ctx, ctx.Repo.TreePath, meta.Size, lfsDataRc)
+		common.ServeContentByReadSeeker(ctx, ctx.Repo.TreePath, lastModified, lfsDataRc)
+		return nil
```

Comment:
Hi! Out of curiousity, why `ServeContentByReadSeeker` and not `ServeContentByReader` here? I understand https://github.com/go-gitea/gitea/pull/24592/files#r1187690002, and reading the code itself, correctly, the latter is the one that supports "Range", so why not use it everywhere? But instead all of the changes use the former, except for the two error cases in the LFS path, which will never return a large file you would need to use "Range:" on anyway.

Reply:
Answering my own question: [http.ServeContent](https://pkg.go.dev/net/http#ServeContent), as used [here](https://github.com/go-gitea/gitea/pull/24592/files#diff-31a0e3934fa53bfa831ed6e3ae64aeeeacf8e9e4eb86b8df836f8f0d9ee493a1R224)

> it handles Range requests properly, sets the MIME type, and handles [...]

But it takes a `ReadSeeker`, so `ServeContentByReader` is reimplementing enough of `ServeContent` to provide the same feature to `Reader`s as well.